gravitee-apim-gateway/gravitee-apim-gateway-tests-sdk/src/test/java/testcases/ManuallyRedeployTestCase.java

```diff
@@ -0,0 +1,216 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.ok;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.gravitee.apim.gateway.tests.sdk.AbstractGatewayTest;
+import io.gravitee.apim.gateway.tests.sdk.annotations.DeployApi;
+import io.gravitee.apim.gateway.tests.sdk.annotations.GatewayTest;
+import io.gravitee.apim.gateway.tests.sdk.annotations.InjectApi;
+import io.gravitee.apim.gateway.tests.sdk.policy.PolicyBuilder;
+import io.gravitee.apim.gateway.tests.sdk.policy.fakes.Header1Policy;
+import io.gravitee.apim.gateway.tests.sdk.policy.fakes.OnRequestPolicy;
+import io.gravitee.apim.gateway.tests.sdk.policy.fakes.Stream1Policy;
+import io.gravitee.apim.gateway.tests.sdk.policy.fakes.Stream2Policy;
+import io.gravitee.definition.model.Api;
+import io.gravitee.gateway.reactor.ReactableApi;
+import io.gravitee.plugin.policy.PolicyPlugin;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.rxjava3.core.http.HttpClient;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@GatewayTest
+@EnableForGatewayTestingExtensionTesting
+public class ManuallyRedeployTestCase extends AbstractGatewayTest {
+
+    public static final String ON_REQUEST_POLICY = "on-request-policy";
+
+    @Test
+    @DisplayName("Should modify and redeploy an API")
+    @DeployApi({ "/apis/success-flow.json" })
+    void shouldRedeployAnApi(HttpClient httpClient, @InjectApi(apiId = "my-api") ReactableApi<?> reactableApi) throws InterruptedException {
+        wiremock.stubFor(get("/team/my_team").willReturn(ok()));
+
+        httpClient
+            .rxRequest(HttpMethod.GET, "/test/my_team")
+            .flatMap(request -> request.rxSend())
+            .flatMapPublisher(response -> {
+                assertThat(response.statusCode()).isEqualTo(200);
+                assertThat(response.headers().contains("X-Gravitee-Policy")).isFalse();
+                return response.toFlowable();
+            })
+            .test()
+            .await()
+            .assertComplete()
+            .assertValue(body -> {
+                assertThat(body.toString()).isEqualTo("OnResponseContent2Policy");
+                return true;
+            })
+            .assertNoErrors();
+
+        if (isLegacyApi(reactableApi.getDefinition().getClass())) {
+            Api api = (Api) reactableApi.getDefinition();
+            api.getProxy().getVirtualHosts().get(0).setPath("/new_test");
+            redeploy(reactableApi);
+        }
```

Comment:
What if it is not legacy API ?

Reply:
here in the test case I just deploy a "legacy" api (not a v4 one) so I do not have to manage especially the v4 one (testable with `isV4Api`)


those methods are useful to distinguish the version of the api and create the good object from definition.
If I had deployed a v4 API, I would have to change the context path through selectors.